test/parallel_api/iterator/zip_iterator.pass.cpp

```diff
@@ -122,6 +124,34 @@ struct test_for_each
     }
 };
 
+struct test_for_each_structured_binding
+{
+    template <typename Policy, typename Iterator1, typename Size>
+    void
+    operator()(Policy&& exec, Iterator1 first1, Iterator1 last1, Size n)
+    {
+        typedef typename ::std::iterator_traits<Iterator1>::value_type T1;
+        auto host_first1 = get_host_pointer(first1);
+
+        auto tuple_first1 = oneapi::dpl::make_zip_iterator(first1, first1);
```

Comment:
Here we have 1st argument equals 2nd argument. Will it be better for test if we have the 2nd argument differed from 1st? Or you are looking for compilation mostly.

Reply:
I think tests should be extended anyway, so I believe that currently I have the very basic test and I would like to agree on the approach for the functionality implementation first.